### PR TITLE
Amplitudes to df (Issue #29)

### PR DESCRIPTION
**Pull request by sboltz**
Wrote an extractor for amplitudes analogous to `picks_to_df`. I also may have gotten a little bit carried away and also made extractors for station magnitudes, magnitudes and arrivals (specifically arrivals for the preferred origin for an event or a user-provided origin to try to avoid some of the mess of trying to sort out which arrivals are real and which are from old rejected origins, but that can be changed to pull them for all origins on an event).

I also made a tweak to `get_reference_time` to make it work if an Origin is supplied and added `get_seed_id` to `obsplus.events.utils`, which will search down through various types of event objects to identify the seed id that is associated with the objects.

I tried to minimize repetition as much as possible, but some of the extractor stuff could probably be improved... to me it seems like they are all similar enough that there should be a way to condense them into one thing.  I'm also not sure how to go about updating the docs.

### CODE REVIEW

**Comment by d-chambers** [comment]
> get_seed_id to obsplus.events.utils, which will search down through various types of event objects to identify the seed id that is associated with the objects.

I love this addition!

**Comment by d-chambers** [comment]
> I tried to minimize repetition as much as possible, but some of the extractor stuff could probably be improved... to me it seems like they are all similar enough that there should be a way to condense them into one thing.

I feel the same way, there should be a more elegant way. I am just not sure yet what it is.

File: tests/test_events/test_seed_id.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+@author: shawn
+"""
+
+import pytest
+
+import obspy
+import obsplus
+
+
+# I'm not sure of the best place to put these tests, feel free to move them
```
Comment (d-chambers) [inline tests/test_events/test_seed_id.py]: I think the end of obsplys/tests/tests_events/test_catalog_utils.py would be the correct place.

File: obsplus/events/utils.py [hunk 2]
```diff
@@ -401,6 +401,50 @@ def get_utc_path(
...
+    # attribute to the shallowest
+    attrs = ["waveform_id", "pick_id", "amplitude_id", "station_magnitude_id"]
+    # Make sure the object has at least one of the required attributes
+    if not len(set(attrs).intersection(set(vars(obj)))):
+        raise TypeError(f"cannot retrieve seed id for objects of type {type(obj)}")
+    # Loop over each of the attributes, if it exists and is not None,
+    # go down a level until it finds a seed id
+    for att in attrs:
+        if hasattr(obj, att):
+            val = getattr(obj, att)
```
[the remark this answers is not in the document]
Comment (sboltz) [reply obsplus/events/utils.py]: Good call. For some reason I didn't realize that could take another argument.

File: obsplus/structures/dfextractor.py [hunk 3]
```diff
@@ -200,6 +200,11 @@ def __call__(self, obj, **kwargs) -> pd.DataFrame:
             # read in any UTCDateTime
             for col in set(iterate(self.utc_columns)) & set(df.columns):
                 df[col] = df[col].apply(_timestampit)
+        #            for col in iterate(self.utc_columns):
```
[the remark this answers is not in the document]
Comment (sboltz) [reply obsplus/structures/dfextractor.py]: Maybe... I will have to take a closer look. I think I added the stuff in comments after running into a problem, and then when I rebased it you had made a change that had fixed the problem that I was encountering and I wanted to keep what I had done to remind myself to look into it.

File: tests/test_events/test_pd_catalog.py [hunk 4]
```diff
@@ -402,6 +583,317 @@ def test_seed_id(self, pick_df):
...
+        assert req_len == len(read_arr_output)
+
+    def test_values(self, read_arr_output, dummy_cat):
+        """ make sure the values of the first arrival are as expected """
+        arr_ser = read_arr_output.iloc[0]
+        origin = dummy_cat[0].preferred_origin()
+        arr = origin.arrivals[0]
+        # Is there a less messy way to check that the function correctly
+        # parsed the amplitude?
+        assert arr_ser["resource_id"] == arr.resource_id.id
```
[the remark this answers is not in the document]
Comment (sboltz) [reply tests/test_events/test_pd_catalog.py]: Yeah, that would definitely be a better way to do that. I found myself cringing while writing these tests.

**Comment by d-chambers** [comment]
Looks good, merging!

**Pull request merged**